### ISSUE TITLE
Successfully trained 355M on a 8GB video card using using 0.4.0 with warnings

### ISSUE THREAD

**Issue by channelz** [minimaxir/aitextgen#96]
Successfully trained 355M on a 8GB video card using using 0.4.0.   

Looks like it wants a good amount of memory otherwise it grabs virtual instead.  For me the process reported 24GB under the  VIRT column using top.

Training took up 7.1GB on the card
Inference took up around 2.5GB

Just some warnings that will pop to be aware of for those attempting: 

`2021-02-26 11:28:54.805677: W tensorflow/core/framework/allocator.cc:107] Allocation of 205852672 exceeds 10% of system memory.`

```
UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
```

**Comment by minimaxir** [comment]
How large was the input dataset? If you had a super large dataset it could take up a lot of system RAM even with encoding.

Those VRAM numbers seem accurate. Colab for 355M results in similar VRAM consumption.

> 2021-02-26 11:28:54.805677: W tensorflow/core/framework/allocator.cc:107] Allocation of 205852672 exceeds 10% of system memory.

That's a VRAM warning (not entirely sure why it's from TensorFlow but OK). Should be fine if training didn't terminate.

> UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")

Meanwhile, this may require a separate ticket (it's gradient-checkpoint specific, enabled by default when fine-tuning the 355M model). Reloading the model fixes the warning (default Colab behavior), not entirely sure why the workaround I tried to use does not work.

**Comment by channelz** [comment]
So everything finished fine.  But yes, gradient-checkpoint seems enabled by default.

**Comment by channelz** [comment]
Upon further examination  The warning below is generated when `generate_sample_text` of the training is called.  The gradient tensor does not need to be calculated when inputs are none as set in line 199 of train.py.  (https://github.com/minimaxir/aitextgen/blob/8dbc36244daf59c04d6a25f95fb409501d4a826d/aitextgen/train.py#L199)

```
UserWarning: None of the inputs have requires_grad=True. Gradients will be None
warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
```
